pycbc/inference/sampler/base.py

```diff
@@ -175,7 +175,7 @@ def setup_output(sampler, output_file):
     # check if we have a good checkpoint and/or backup file
     logging.info("Looking for checkpoint file")
     checkpoint_valid = validate_checkpoint_files(checkpoint_file,
-                                                 backup_file)
+                                                 backup_file, sampler.name)
```

Comment:
Continuing my suggestion above, add a `check_nsamples` keyword argument to `setup_output` (with default True), which you then pass to `validate_checkpoint_files`.